### PR TITLE
Add USE_SYSTEM_DEPS flag and install target

### PR DESCRIPTION
**Pull request by otreblan**
If its `-DUSE_SYSTEM_DEPS=ON` the deps folder is ignored and only cppglob is linked.
Else the libraries at deps are build and linked.

Also added the `deps/cxxopts/include` to the include directories
because in Arch linux, cxxopts puts the `cxxopts.hpp` header in `/usr/include/`

### CODE REVIEW

**Comment by otreblan** [comment]
This PR is needed to build the aur package.
https://aur.archlinux.org/packages/libtree-git/
(The PKGBUILD is using my fork as source)

**Comment by maxice8** [comment]
Please add `fixes #5` to your `Add install target` commit

**Comment by maxice8** [comment]
With this patch i get the following:

```
[ 14%] Building CXX object CMakeFiles/libtree.dir/src/main.cpp.o
/home/builder/aports/testing/libtree/src/libtree-1.0.0/src/main.cpp:3:10: fatal error: cxxopts.hpp: No such file or directory
    3 | #include <cxxopts.hpp>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/libtree.dir/build.make:83: CMakeFiles/libtree.dir/src/main.cpp.o] Error 1
make[2]: Leaving directory '/home/builder/aports/testing/libtree/src/libtree-1.0.0/build'
make[1]: *** [CMakeFiles/Makefile2:96: CMakeFiles/libtree.dir/all] Error 2
make[1]: Leaving directory '/home/builder/aports/testing/libtree/src/libtree-1.0.0/build'
make: *** [Makefile:150: all] Error 2
make: Leaving directory '/home/builder/aports/testing/libtree/src/libtree-1.0.0/build'
>>> ERROR: libtree: build failed
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -19,7 +26,21 @@ set(SOURCE
 )
 
 add_executable(libtree ${SOURCE})
-target_link_libraries(libtree cxxopts termcolor cppglob ELFIO)
-target_include_directories(libtree PRIVATE include deps/cppglob/include ${CMAKE_CURRENT_BINARY_DIR})
+target_include_directories(libtree PRIVATE
+    include
+    deps/cppglob/include
+    deps/cxxopts/include
```
Comment (maxice8) [inline CMakeLists.txt:32]: possibly also missing deps/termcolor/include ?

Comment (otreblan) [reply CMakeLists.txt]: Solved

**Comment by otreblan** [comment]
Closes #5

File: CMakeLists.txt [hunk 2]
```diff
@@ -19,7 +26,25 @@ set(SOURCE
...
+    ${CMAKE_CURRENT_BINARY_DIR}
+)
+
+if(USE_SYSTEM_DEPS)
+    target_link_libraries(libtree cppglob)
+else()
+    target_link_libraries(libtree cxxopts termcolor cppglob ELFIO)
+    target_include_directories(libtree PRIVATE
+        deps/cppglob/include
+        deps/termcolors/termcolors
```
Comment (maxice8) [inline CMakeLists.txt:40]: isn't it deps/termcolors/include/termcolors ?

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,13 +1,20 @@
 cmake_minimum_required (VERSION 3.0)
 project (libtree)
 
+include(GNUInstallDirs)
+
+set(CACHE BOOL USE_SYSTEM_DEPS OFF)
```
Comment (maxice8) [inline CMakeLists.txt:6]: Wouldn't it be better to use an option for the user ?

```cmake
option(USE_SYSTEM_DEPS "Use system cppglob" OFF)
```

File: CMakeLists.txt (hunk 2 above)
Comment (otreblan) [reply CMakeLists.txt:40]: No.
The header its included like this `#include <termcolor/termcolor.hpp>`

File: CMakeLists.txt (hunk 3 above)
Comment (otreblan) [reply CMakeLists.txt]: ok

**Comment by haampie** [comment]
Thanks a lot, this is very useful. I've removed the cppglob targets from installing as well. I think the CMakeLists.txt could be simplified a bit if upstream they remove the `BUILD_SHARED` / `BUILD_STATIC` options and instead use smth like `BUILD_SHARED_LIBS=ON/OFF` with a single target.

**Comment by haampie** [comment]
@otreblan could you maybe update the URL of your AUR package? :)

**Comment by otreblan** [comment]
> @otreblan could you maybe update the URL of your AUR package? :)

Done

**Comment by haampie** [comment]
Thanks!!

**Pull request merged**